### ISSUE TITLE
Improve readability of SI format for very small/large numbers

### ISSUE THREAD

**Issue by patkwee** [d3/d3-format#111]
It would be great to improve the 'human-readability' of SI format "s" for very small/large numbers. At the moment d3-format generates:

```javascript
console.log(format("s")(1e-50))  -> "0.00000000000000000000000001y"
console.log(format("s")(1e+50)) -> "100000000000000000000000000Y"
```

I would propose to fall back to exponent notation ("e" format) for numbers below `abs(x)<1e-24` or above `abs(x)>1e24`.

In my application, usually the numbers are in a range where the "s"-format makes sense. Though, sometimes, e.g. due to numerical inaccuracy of calculations, I get very small numbers, that blow-up my output.

**Comment by Fil** [comment]
For small numbers you could round the number before formatting: 
> a => format("s")(+a.toFixed(26));

**Comment by patkwee** [comment]
Yes, true. I currently just check for `abs(x)<1e-24` and then use "e" or "s" formatting. My point is more that a 'human' would format 1e-50 rather as `1e-50` than as `0.00000000000000000000000001y` and therefore it might be an interesting improvement proposal.

Libraries like vega and vega-lite also use d3-format. There a workaround is also possible with custom formatters, but it gets more and more tedious.
